app/routes/resources+/theme-switch.tsx

```diff
@@ -0,0 +1,114 @@
+import { useForm, getFormProps } from '@conform-to/react'
+import { parseWithZod } from '@conform-to/zod'
+import { invariantResponse } from '@epic-web/invariant'
+import {
+	json,
+	type ActionFunctionArgs,
+} from '@remix-run/node'
+import {
+	useFetcher,
+	useFetchers,
+} from '@remix-run/react'
+import { z } from 'zod'
+import { Icon } from '#app/components/ui/icon.tsx'
+import { useHints } from '#app/utils/client-hints.tsx'
+import { useRequestInfo } from '#app/utils/request-info.ts'
+import { type Theme, setTheme } from '#app/utils/theme.server.ts'
+
+const ThemeFormSchema = z.object({
+	theme: z.enum(['system', 'light', 'dark']),
+})
+
+export async function action({ request }: ActionFunctionArgs) {
```

Comment:
This was copied over, but the original action in the root was not removed.

Reply:
ah good catch, thanks.. removed!